framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -266,8 +344,16 @@ private static boolean isIntLiteral(String s, FlowExpressionContext context) {
         if (context.parsingMember) {
             return false;
         }
-        Matcher intMatcher = INT_PATTERN.matcher(s);
-        return intMatcher.matches();
+        try {
+            Expression e = parseExpression(s);
```

Comment:
The next 6 lines can be replaced by
`return parseExpression(s).getClass().equals(IntegerLiteralExpr.class)`

In general, never do
```
if (p) {
  return true;
} else {
  return false;
}
```
Instead, just do `return p;`

The same comment applies to several other blocks of code.